test/Transforms/canonicalize.mlir

```diff
@@ -261,16 +261,21 @@ func @memref_cast_folding(%arg0: memref<4 x f32>, %arg1: f32) -> f32 {
   // CHECK-NEXT: %c0 = constant 0 : index
   %c0 = constant 0 : index
 
+  %dim = dim %1, 0 : memref<? x f32>
+
+  // CHECK-NEXT: affine.load %arg0[3]
```

Comment:
This is not passing presubmit testing (seems you are missing "constant 4" op in between previous check and next). Could you verify locally?

Reply:
You are right (this was because my local copy had another pending pull request, and the further simplification was a combined effect of both). Fixed now.